### PR TITLE
Update translation of 1-js/04-object-basics/04-object-methods

### PR DESCRIPTION
**Pull request by leviding**
#### 所做更改如下：

文件名 | 参考上游 commit | 更改（理由）
-|-|-
article.md | https://github.com/javascript-tutorial/en.javascript.info/commit/9fd644cc839f4ce660a128481f315691cebc7782 | 更新译文
2-check-syntax/solution.md | https://github.com/javascript-tutorial/en.javascript.info/commit/a95a2f5725c4f7f5984ac14d0acfd47d082210a7 | 更新译文
3-why-this/solution.md | https://github.com/javascript-tutorial/en.javascript.info/commit/9ad9063d00dd7bf7a5e993cb9bded0d4e0cdf9b0 | 更新译文
3-why-this/task.md | https://github.com/javascript-tutorial/en.javascript.info/commit/dd52fb91cdbbbb274383b5f81576c2effbc70797 | 更新译文
4-object-property-this/solution.md | https://github.com/javascript-tutorial/en.javascript.info/commit/0de5046a40bee9f0855e1a09f38a6acf7a5e08db | 更新译文
4-object-property-this/task.md | https://github.com/javascript-tutorial/en.javascript.info/commit/9ad9063d00dd7bf7a5e993cb9bded0d4e0cdf9b0 | 更新译文
7-calculator/solution.md | https://github.com/javascript-tutorial/en.javascript.info/commit/408ba7d4e4d5a66c9ae4ce1d236b407da0d67ef2 | 更新译文
7-calculator/task.md | https://github.com/javascript-tutorial/en.javascript.info/commit/9ad9063d00dd7bf7a5e993cb9bded0d4e0cdf9b0 | 更新译文
8-chain-calls/task.md | https://github.com/javascript-tutorial/en.javascript.info/commit/397a3aff4efe1f88b4af389d72a3e186fb10004e | 更新译文
8-chain-calls/task.md | https://github.com/javascript-tutorial/en.javascript.info/commit/408ba7d4e4d5a66c9ae4ce1d236b407da0d67ef2#diff-533794fc2283ec2cab0c9b542d8df826 | 更新译文

### CODE REVIEW

File: 1-js/04-object-basics/04-object-methods/article.md [hunk 1]
```diff
@@ -157,28 +158,28 @@ let user = {
...
 ```
 
-如果在 `alert` 中以 `this.name` 替换 `user.name`，那么代码就会正常运行。
+如果我们在 `alert` 中以 `this.name` 替换 `user.name`，那么代码就会正常运行。
 
-## “this” 不受限制
+## "this" 不受限制
 
-在 JavaScript 中，"this" 关键字与大多数其他编程语言中的不同。首先，它可以用于任何函数。
+在 JavaScript 中，"this" 关键字与其他大多数编程语言中的不同。JavaScript 中的 this 可以用于任何函数。
```
Comment (MartinsYong) [inline 1-js/04-object-basics/04-object-methods/article.md]:
```suggestion
在 JavaScript 中，`this` 关键字与其他大多数编程语言中的不同。JavaScript 中的 `this` 可以用于任何函数。
```

File: 1-js/04-object-basics/04-object-methods/article.md [hunk 2]
```diff
@@ -212,31 +216,32 @@ function sayHi() {
...
+通常这种调用是程序出错了。如果在一个函数内部有 `this`，那么通常意味着它是在对象上下文环境中被调用的。
+````
 
-```smart header="The consequences of unbound `this`"
-如果你来自其他的编程语言，那么你可能熟悉『绑定 `this`』的概念。在对象定义的方法中，`this` 总是指向该对象。
+```smart header="解除 `this` 绑定的后果"
+如果你经常使用其他的编程语言，那么你可能已经习惯了“绑定 `this`”的概念，即在对象中定义的方法总是有指向该对象的 `this`。
 
-在 JavaScript 中，`this` 是『自由』的，它的值是在调用时进行求值的，它的值并不取决于方法声明的位置，而是（取决）于在『点之前』的是什么对象。
+在 JavaScript 中，`this` 是“由“”的，它的值是在调用时计算出来的，它的值并不取决于方法声明的位置，而是取决于在“点符号前”的是什么对象。
```
Comment (MartinsYong) [inline 1-js/04-object-basics/04-object-methods/article.md]:
```suggestion
在 JavaScript 中，`this` 是“自由”的，它的值是在调用时计算出来的，它的值并不取决于方法声明的位置，而是取决于在“点符号前”的是什么对象。
```

File: 1-js/04-object-basics/04-object-methods/article.md [hunk 3]
```diff
@@ -293,36 +298,38 @@ hi(); // 错误，因为 this 未定义
...
 ```
 
-这里 `hi = user.hi` 把函数赋值给变量，其后的最后一行是完全独立的，所以它没有 `this`。
+这里 `hi = user.hi` 把函数赋值给变量，其后的最后一行代码是完全独立的，所以它没有 `this`。
 
-**为了让 `user.hi()` 有效，JavaScript 用一个技巧 —— 这个 `'.'` 点返回的不是一个函数，而是一种特殊的[引用类型](https://tc39.github.io/ecma262/#sec-reference-specification-type)的值。**
+**为了让 `user.hi()` 有效，JavaScript 用了一个技巧 —— 这个 `'.'` 点符号返回的不是一个函数，而是一种特殊的 [引用类型](https://tc39.github.io/ecma262/#sec-reference-specification-type) 的值。**
 
-引用类型是一种『规范中有的类型』。我们不能明确地指定它，但是可以在语言内部使用。
+引用类型是一种“规范中有的型“”。我们不能明确地指定它，但它被用在编程语言的内部。
```
Comment (MartinsYong) [inline 1-js/04-object-basics/04-object-methods/article.md]:
```suggestion
引用类型是一种“规范中有的型”。我们不能明确地指定它，但它被用在编程语言的内部。
```

**Comment by leviding** [comment]
@MartinsYong Thanks a lot！

**Pull request merged**